src/samVisualize/graphGeneration/graph.ts

```diff
@@ -44,52 +46,84 @@ export class Graph {
 
     /**
      * Stores which nodes exist in the graph
+     *
      * Node names are used as keys.
+     *
      * Exists to avoid iteration over the graph.nodes list to check for duplicates
      */
     private readonly nodeSet: Set<string>
 
     /**
-     * Stores which links exist in the graph
-     * The concatenation of the Link source, target, and type strings are used as keys because
-     * neither source, target, nor type properties are individually unique.
-     * The combination of all three properties determines a unique Link.
+     * Stores which links exist in the graph.
+     *
+     * Uses a Map instead of a Set to store the type of link,
+     * to allow DependsOn links between two nodes to be replaced by an Intrinsic Function link between the same nodes.
+     *
+     * The concatenation of the Link source and target strings are used as keys because
+     * neither source nor target properties are individually unique.
+     * The combination of both properties determines a unique Link.
+     *
      * Exists to avoid iteration over the graph.links list to check for duplicates
      */
-    private readonly linkSet: Set<string>
+    private readonly linkSet: Map<string, string>
```

Comment:
since it is no longer a set, do we need to change the name?

Reply:
Changed for clarity